### PR TITLE
v6.0.0 alpha.3

### PR DESCRIPTION
**Pull request by MBilalShafi**
- [x] Compare the last tag with the branch upon which you want to release (`next` for the alpha / beta releases and `master` for the current stable version).
      To do so, use `yarn release:changelog` The options are the following:

  ```bash
  yarn release:changelog
     --githubToken   YOUR_GITHUB_TOKEN (needs "public_repo" permission)
     --lastRelease   The release to compare against (default: the last one)
     --release       The branch to release (default: master)
  ```

  You can also provide the github token by setting `process.env.GITHUB_TOKEN` variable.

  In case of a problem, another method to generate the changelog is available at the end of this page.

- [x] Clean the generated changelog, to match the format of [https://github.com/mui/mui-x/releases](https://github.com/mui/mui-x/releases).
- [x] Update the root `package.json`'s version
- [x] Update the versions of the other `package.json` files and of the dependencies with `yarn release:version`.
- [x] Fix manually the package version in `x-date-pickers/package.json` and `x-date-pickers-pro/package.json`.
- [x] Open PR with changes and wait for review and green CI.
- [ ] Merge PR once CI is green, and it has been approved.

### Release the packages

- [ ] Checkout the last version of the working branch
- [ ] Make sure you have the latest dependencies installed: `yarn`.
- [ ] Build the packages: `yarn release:build`.
- [ ] Release the versions on npm: `yarn release:publish` (you need your 2FA device).
- [ ] Push the newly created tag: `yarn release:tag`.

### Publish the documentation

The documentation must be updated on the `docs-vX` branch (`docs-v4` for `v4.X` releases, `docs-v5` for `v5.X` releases, ...)

- [ ] Push the working branch on the documentation release branch to deploy the documentation with the latest changes.

```sh
git push upstream next:docs-next -f
```

You can follow the deployment process [on the Netlify Dashboard](https://app.netlify.com/sites/material-ui-x/deploys?filter=docs-v5)
Once deployed, it will be accessible at https://material-ui-x.netlify.app/ for the `docs-v5` deployment.

### Publish GitHub release

- [ ] After documentation is deployed, publish a new release on [GitHub releases page](https://github.com/mui/mui-x/releases)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,181 @@
...
+
+_Oct 13, 2022_
+
+We'd like to offer a big thanks to the 8 contributors who made this release possible. Here are some highlights ✨:
+
+
+- 📚 Documentation improvements
+
+From https://github.com/mui/mui-x/pull/6381
+
```
Comment (alexfauquette) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,181 @@
...
+
+#### Breaking changes
+
+- All the props used by the mobile and desktop wrappers to override components or components props has been replaced by component slots. You can find more information about this pattern in the [MUI Base documentation](https://mui.com/base/getting-started/usage/#shared-props).
```
Comment (alexfauquette) [inline CHANGELOG.md:32]:
```suggestion
- All the props used by the mobile and desktop wrappers to override components or components' props have been replaced by component slots. You can find more information about this pattern in the [MUI Base documentation](https://mui.com/base/getting-started/usage/#shared-props).
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,181 @@
...
+
+  Some of the names have also been prefixed by `desktop` when it was unclear that the behavior was only applied on the desktop version of the pickers (or the responsive version when used on a desktop).
+
+  The `DialogProps` prop has been replaced by a `dialog` component props slot on a responsive an mobile pickers:
```
Comment (alexfauquette) [inline CHANGELOG.md:36]:
```suggestion
  The `DialogProps` prop has been replaced by a `dialog` component props slot on responsive and mobile pickers:
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,181 @@
...
+  // Same on DesktopDatePicker, DateTimePicker, DesktopDateTimePicker, 
+  // TimePicker, DesktopTimePicker, DateRangePicker and DesktopDateRangePicker.
+  <DatePicker
+  -  TrapFocusProps={{ isEnabled: () => false }}
+  +  componentsProps={{ desktopTrapFocus: { isEnabled: () => false }}}
+  />
+  ```
+ 
+From https://github.com/mui/mui-x/pull/6389
+
```
Comment (alexfauquette) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,177 @@
...
+- [core] Add CodeQL workflow (#6387) @DanailH
+- [core] Add missing breaking change to the changelog (#6471) @flaviendelangle
+- [core] Fix playground structure (#6466) @LukasTy
+- [core] Fix tests for pasting on fields (#6465) @flaviendelangle
+- [core] Remove absolute link (#6420) @flaviendelangle
+- [core] Remove unused `react-text-mask` package (#6408) @LukasTy
+- [core] Send explicit warning when dayjs locale is not found (#6424) @alexfauquette
+- [core] Test validation on textfield and date views (#6265) @alexfauquette
+- New components: `MultiInputDateTimeRangePicker` and `MultiInputTimeRangePicker` (#6392) @alexfauquette
```
Comment (alexfauquette) [inline CHANGELOG.md:174]:
```suggestion
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,177 @@
...
+  ```diff
+  -MuiYearPicker: {
+  +MuiYearCalendar: {
+  ```
+
+#### Changes
+
+- [DatePicker] Allows to fix the number of week displayed (#6299) @alexfauquette
+- [DateRangePicker] Fix calendar day outside of month layout shifting on hover (#6448) @alexfauquette
+- [fields] Prepare the field exports for the public release (#6467) @flaviendelangle
```
Comment (alexfauquette) [inline CHANGELOG.md:148]:
```suggestion
- [fields] New components: `MultiInputDateTimeRangePicker` and `MultiInputTimeRangePicker` (#6392) @alexfauquette
- [fields] Prepare the field exports for the public release (#6467) @flaviendelangle
```

File: CHANGELOG.md (hunk 5 above)
Comment (alexfauquette) [reply CHANGELOG.md:174]: Sorry, wrong PR title 🙈

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,177 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## v6.0.0-alpha.3
+
+_Oct 13, 2022_
+
+We'd like to offer a big thanks to the 8 contributors who made this release possible. Here are some highlights ✨:
+
+- 🚀 Simplify `useGridApiContext` and `useGridApiRef` type overrides (#6423) @cherniavskii
```
Comment (MBilalShafi) [inline CHANGELOG.md:12]: Let me know if we should highlight something else too?

Comment (cherniavskii) [reply CHANGELOG.md:12]: I wouldn't highlight #6423, since it's an internal change and doesn't impact the users

Comment (flaviendelangle) [inline CHANGELOG.md:12]: We clearly need a highlight for the new fields

@joserodolfofreitas @LukasTy @alexfauquette 

First draft:

- ⌚️ New components to edit dates and times with the beyboard, without using any modal or dropdown UI.
  Please check out our [documentation](https://next.mui.com/x/react-date-pickers/fields/) to discover those new components.

  - [`DateField`](https://next.mui.com/x/react-date-pickers/date-field/) to edit a single date
  - [`TimeField`](https://next.mui.com/x/react-date-pickers/time-field/)  to edit a single time
  - [`DateTimeField`](https://next.mui.com/x/react-date-pickers/date-time-field/)  to edit a single date and time
  - [`MultiInputDateRangeField` / `SingleInputDateRangeField`](https://next.mui.com/x/react-date-pickers/date-range-field/) to edit a date range with two inputs
  - [`MultiInputTimeRangeField`](https://next.mui.com/x/react-date-pickers/time-range-field/) to edit a time range with two inputs
  - [`MultiInputDateTimeRangeField`](https://next.mui.com/x/react-date-pickers/date-time-range-field/) to edit a date time range with two inputs
  
  ⚠️ These components are unstable.
  They might receive breaking changes on their props to have the best components possible by the time of the stable release.

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,177 @@
...
+- 📝 Allow to limit to one filter per column for `DataGridPro` and `DataGridPremium` (#6333) @MBilalShafi
+- 🎁 Support pasting in pickers @flaviendelangle
+- 📚 Documentation improvements
+- 🐞 Bugfixes
+
+### `@mui/x-data-grid@v6.0.0-alpha.3` / `@mui/x-data-grid-pro@v6.0.0-alpha.3` / `@mui/x-data-grid-premium@v6.0.0-alpha.3`
+
+#### Changes
```
[the remark this answers is not in the document]
Comment (flaviendelangle) [reply CHANGELOG.md:20]: To have the breaking change automatically copied here (and avoid mistakes), I do the following structure in your PR description

```
## Changelog

#### Breaking changes
```

File: CHANGELOG.md (hunk 7 above)
Comment (LukasTy) [reply CHANGELOG.md:12]: Minor tweaking:

- ⌚️ New components to edit date and time with <kbd>keyboard</kbd>—without using any modal or dropdown UI.
  Please check out our [documentation](https://next.mui.com/x/react-date-pickers/fields/) to discover those new components.

  - [`DateField`](https://next.mui.com/x/react-date-pickers/date-field/) to edit date
  - [`TimeField`](https://next.mui.com/x/react-date-pickers/time-field/)  to edit time
  - [`DateTimeField`](https://next.mui.com/x/react-date-pickers/date-time-field/)  to edit date and time
  - [`MultiInputDateRangeField` / `SingleInputDateRangeField`](https://next.mui.com/x/react-date-pickers/date-range-field/) to edit date range
  - [`MultiInputTimeRangeField`](https://next.mui.com/x/react-date-pickers/time-range-field/) to edit time range with two inputs
  - [`MultiInputDateTimeRangeField`](https://next.mui.com/x/react-date-pickers/date-time-range-field/) to edit date and time range with two inputs
  
  ⚠️ These components are unstable.
  They might receive breaking changes on their props to have the best components possible by the time of the stable release.

**Comment by flaviendelangle** [comment]
> We need #6480 before the release +1

Merged

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,211 @@
...
+  - [`MultiInputDateRangeField` / `SingleInputDateRangeField`](https://next.mui.com/x/react-date-pickers/date-range-field/) to edit date range
+  - [`MultiInputTimeRangeField`](https://next.mui.com/x/react-date-pickers/time-range-field/) to edit time range with two inputs
+  - [`MultiInputDateTimeRangeField`](https://next.mui.com/x/react-date-pickers/date-time-range-field/) to edit date and time range with two inputs
+  
+  ⚠️ These components are unstable.
+  They might receive breaking changes on their props to have the best components possible by the time of the stable release.
+
+- 📝 Allow to limit to one filter per column for `DataGridPro` and `DataGridPremium` (#6333) @MBilalShafi
+- 🎁 Support pasting in pickers @flaviendelangle
+- 📚 Documentation improvements
```
Comment (flaviendelangle) [inline CHANGELOG.md:26]: Should we add a line for the new validation page ? 
We added one for the `apiRef` page a few months back

```suggestion
- 📚 New [page describing the validation props on each picker](https://next.mui.com/x/react-date-pickers/validation/) (#6064) @flaviendelangle

- 📚 Documentation improvements
```

File: CHANGELOG.md (hunk 7 above)
Comment (MBilalShafi) [reply CHANGELOG.md]: Added

File: CHANGELOG.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (MBilalShafi) [reply CHANGELOG.md:54]: Breaking changes added

File: CHANGELOG.md (hunk 6 above)
Comment (MBilalShafi) [reply CHANGELOG.md:183]: Updated

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,6 +3,211 @@
...
+- [pickers] Replace the `Picker` prefix in the view component by `Calendar` (eg: `MonthPicker` => `MonthCalendar`) (#6389) @flaviendelangle
+- [pickers] Support pasting on fields (#6364) @flaviendelangle
+- [pickers] Use slots in the mobile and desktop wrappers instead of `XXXComponent` and `XXXProps` (#6381) @flaviendelangle
+
+### Docs
+
+- [docs] Add migration to DataGrid v6 page (#6235) @m4theushw
+- [docs] Create first publishable version of the field doc (#6323) @flaviendelangle
+- [docs] Fix trailing spaces in the readme @oliviertassinari
+- [docs] New page for the pickers: Validation (#6064) @flaviendelangle
```
Comment (flaviendelangle) [inline CHANGELOG.md:197]:
```suggestion
- [docs] New page for the pickers: Validation (#6064) @flaviendelangle
- [docs] Organize migration pages (#6480) @flaviendelangle
```

**Review by flaviendelangle — APPROVED** [review]

**Review by alexfauquette — APPROVED** [review]

**Pull request merged**